pytorch_to_returnn/torch/nn/modules/operator.py

```diff
@@ -22,7 +23,21 @@ class Range(Module):
   is_original_torch_module = False
 
   def create_returnn_layer_dict(self, limit, start, delta, dtype, sparse=False) -> Dict[str, Any]:
-    return {"class": "range", "limit": limit, "start": start, "delta": delta, "dtype": dtype, "sparse": sparse}
+    if isinstance(limit, Tensor):
+      return {"class": "range_from_length", "from": self._get_input_layer_name(limit)}
+    else:
+      assert isinstance(limit, int)
+      return {"class": "range", "limit": limit, "start": start, "delta": delta, "dtype": dtype, "sparse": sparse}
+
+  def _get_output_shape_from_returnn(self,
+                                     inputs_flat: List[Optional[Union[Tensor, int, bool]]], layer: LayerBase
+                                     ) -> Tuple[Tuple[int, ...], Dict[int, int]]:
+    limit, start, delta, *_ = inputs_flat
+    if isinstance(limit, Tensor):
+      limit = limit.numpy()
```

Comment:
This might need some extra error check to catch an error earlier on that this value is known. Maybe we need to introduce a new flag `Tensor.is_defined` or so. If this is not known, this could be due to a bug (your current case).

But in principle, when it (`limit`) is really dynamic, the question is what we should do then. I don't see a good solution. Maybe we would just not allow this.

Reply:
Yes not allowing it for now seems reasonable to me. The PR is already diverging significantly from what we originally started it for.

Reply:
How would this flag `Tensor.is_defined` be handled? Every time there is a module which does not take care of the numpy buffer it would be set to `False` and subsequently remain so?

Reply:
And here in this code, you would just make:
```python
assert limit.is_defined
```

Reply:
And the `False` would be set in `Module.make_output_tensor_from_returnn`?

Reply:
`is_defined = numpy_buffer is not None` sets both True and False.